src/App.tsx

```diff
@@ -29,67 +30,121 @@ type ReorderOptions = {
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const {
+      LENGTH,
+      ALPHABET,
+      NONE,
+    } = SortType;
+    const visibleReset = (isReversed || sortType !== NONE);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info', {
+              'is-light': sortType !== ALPHABET,
+            })}
+            onClick={() => (this.setState({ sortType: ALPHABET }))}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success', {
+              'is-light': sortType !== LENGTH,
+            })}
+            onClick={() => (this.setState({ sortType: LENGTH }))}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={() => (
+              isReversed
+                ? this.setState({ isReversed: false })
+                : this.setState({ isReversed: true })
+            )}
+          >
+            Reverse
+          </button>
+
+          {visibleReset && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => (this.setState({
+                isReversed: false,
+                sortType: NONE,
+              }))}
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {getReorderedGoods(
+              goodsFromServer,
+              this.state,
+            ).map((good, index) => (
+              <li key={good + index.toString()} data-cy="Good">{good}</li>
```

Comment:
okay, but no need to use toString, everything that is added to string  will be converted to string

Reply:
![image](https://user-images.githubusercontent.com/84275496/197146645-9ed4477f-3f15-47e0-a8b1-5285c799c8fb.png)

Reply:
funny picture, but i must use String() or toString() cause linter spams mistake

Reply:
oh, ok then